### ISSUE TITLE
Add `DashboardLoaded` event for lazy loading

### ISSUE THREAD

**Issue by AlexvZyl** [nvimdev/dashboard-nvim#368]
**Is your feature request related to a problem? Please describe.**
I am using lazy.nvim to load my plugins.  Something that would be really nice, is if I can first load dashboard (so that it appears super fast) and then after load some extra plugins.

**Describe the solution you'd like**
Fire an event (`DashboardLoaded` or similar) when dashboard is done rendering.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by glepnir** [comment]
in 2019  vimscript version have this event 😄  .  I am  rewriting in new version split theme and provide interface for custom theme easily .

**Comment by AlexvZyl** [comment]
@glepnir 45ea1f1e9c469f88e47e56f1b312aea027e1ba68 👀

**Comment by glepnir** [comment]
oh shit there has 17 open issue ..sry recently most work on neovim core and I don't want write lua recently :vomiting_face:

**Comment by AlexvZyl** [comment]
It seems to me that the even fires before the dashboard is rendered.

**Comment by AlexvZyl** [comment]
> oh shit there has 17 open issue ..sry recently most work on neovim core and I don't want write lua recently :vomiting_face: 

You don't like Lua?

**Comment by glepnir** [comment]
It’s just that I have written too much Lua recently. Need to write some other language.

> It seems to me that the even fires before the dashboard is rendered.

what ? wired. it should ran into next event loop.

**Comment by glepnir** [comment]
if you are using lazy. lazy `event` should check has `User` with pattern. otherwise it not works probably.

**Comment by AlexvZyl** [comment]
> if you are using lazy. lazy `event` should check has `User` with pattern. otherwise it not works probably.

Yes, I did this.  When I am back at my desk I will check if I can see what is going on.

**Comment by glepnir** [comment]
lazy event should do something like `event:find('User') then  vim.split(event, '%s') ` then do exec autocmds with `User` and pattern .

**Comment by AlexvZyl** [comment]
When the event fires, the dashboard is at this stage:

![Screenshot 2023-09-19 at 18 05 44](https://github.com/nvimdev/dashboard-nvim/assets/81622310/d416a5d4-43e0-48a0-9833-812213b9a4e7)

**Comment by AlexvZyl** [comment]
This stil occurs even when I remove dashboard's cache.  Though a large cache might cause the dashboard to be rendered in the wrong event loop.
